src/Company/Yurtici/Service.php

```diff
@@ -0,0 +1,222 @@
+    {
+        $shipmentStatus = new ShipmentStatus();
+        $shipmentStatus->setReferenceNumber($referenceNumber);
+        $res = $this->YKClient->queryShipment($referenceNumber);
+        $statusArray = $this->isErrorMessage($res, $shipmentStatus);
+        if ($statusArray[1]) {
+            return $statusArray[0];
+        }
+        $shipmentStatus = $statusArray[0];
+        // set the status code
+        if (property_exists($res->ShippingDeliveryVO->shippingDeliveryDetailVO, 'operationCode')) {
+            $opCode = $res->ShippingDeliveryVO->shippingDeliveryDetailVO->operationCode;
+            $opMsg = $res->ShippingDeliveryVO->shippingDeliveryDetailVO->operationMessage;
+            $shipmentStatus->setStatusDetails($opMsg);
+            $shipmentStatus->setOriginalStatus($opCode);
+            $shipmentStatus->setStatusCode($this->mapStatus($opCode));
+        }
+        // makes sure there is such a field
+        if (!property_exists($res->ShippingDeliveryVO->shippingDeliveryDetailVO, 'shippingDeliveryItemDetailVO')) {
+            return $shipmentStatus;
+        }
+        // Update Time from document Date and Document Time
+        if (property_exists($res->ShippingDeliveryVO->shippingDeliveryDetailVO->shippingDeliveryItemDetailVO, 'documentDate')) {
+            $date = $res->ShippingDeliveryVO->shippingDeliveryDetailVO->shippingDeliveryItemDetailVO->documentDate;
+            $time = $res->ShippingDeliveryVO->shippingDeliveryDetailVO->shippingDeliveryItemDetailVO->documentTime;
+            $shipmentStatus->setUpdateTime(\DateTime::createFromFormat("YmdGis", $date . $time, new \DateTimeZone('Asia/Istanbul')));
+        }
+
+        return $shipmentStatus;
+    }
+
+    /**
+     * @param string[] $list
+     *
+     * @return ShipmentStatus[]
+     */
+    public function getShipmentStatusByReferenceNumberList(array $list): array
+    {
+        $shipmentStatusList = array();
+        foreach ($list as $value) {
+            $shipmentStatus = $this->getShipmentStatusByReferenceNumber($value);
+            $shipmentStatusList[] = $shipmentStatus;
+        }
+        return $shipmentStatusList;
+    }
+
+
+    /**
+     * Takes the soap client response and processes it until TrackingNumber fullfilment
+     * for both ShipmentStatis and PackageInfo types
+     * 
+     * @param object $res takes the soap client response as $res
+     * @param ShipmentStatus,PackageInfo $package it takes ShipmentStatus
+     * or PackageInfo as parameter
+     * @return array 
+     * the first index is the processed item
+     * the second index is a flag for package to be processed more
+     */
+    private function isErrorMessage($res, $packageInfo)
+    {
+        if ($res->ShippingDeliveryVO->outResult != "Başarılı") {
+            // error
+            $packageInfo->setErrorMessage($res->ShippingDeliveryVO->outResult);
+            $packageInfo->setErrorCode($res->ShippingDeliveryVO->errCode);
+            return [$packageInfo, true];
```

Comment:
response içinde @packageInfo'yu dönmene gerek yok sanırım. Gönderdiğin parametre önceki objeye link olduğu için bu fonksiyonda çağırdığın o objeye ait metodlar aynı datayı değiştirecektir ve önceki metodda kullanabilirsin.